### PR TITLE
Upgrade putfile logs

### LINKED ISSUE #1359: Update putfile -i logging
So that we output a log for each file we read from the URL and put to pachyderm. Otherwise for large data sets, we don't have any real way to measure progress during ingestion.

### PR DESCRIPTION
**Pull request by sjezewski**
Fixes #1359 

For every object that we're putting from an obj store into pachyderm, it should not get its own log line. This will give us a breadcrumb trail of progress for large data set uploads.

Here's an example (expanded) log line thats generated from a recursive putfile now:

```
2017-02-25T01:26:40Z INFO  
{
    "duration": "208.555745ms",
    "filepath": "test/20006.txt",
    "method": "putFileObjHelper",
    "objpath": "test/20006.txt",
    "request": "file:<commit:<repo:<name:\"debug\" > id:\"i\" > path:\"test\" > file_type:FILE_TYPE_REGULAR url:\"s3://tests3recursiveputs/test\" recursive:true ",
    "service": "pfs.API"
}
```

### CODE REVIEW

File: src/server/pfs/server/api_server.go [hunk 1]
```diff
@@ -318,8 +320,50 @@ func (a *apiServer) PutFile(putFileServer pfs.API_PutFileServer) (retErr error)
...
+	l := &putFileHelperLog{
+		Service:  "pfs.API",
+		Method:   "putFileObjHelper",
+		Request:  requestString,
+		Duration: duration.String(),
+		Error:    err,
+		FilePath: filePath,
+		ObjPath:  objPath,
+	}
+	logJSON, err := json.Marshal(l)
```
Comment (jdoliner) [inline src/server/pfs/server/api_server.go]: I don't think you need to be serializing this, the whole point of `lion` is that it should be able to accept your log structures directly and do the serializing itself.

Comment (sjezewski) [inline src/server/pfs/server/api_server.go]: Just supplying the struct results in the following log line:

```
2017-02-27T18:06:47Z INFO  &{pfs.API putFileObjHelper file:<commit:<repo:<name:"debug" > id:"j" > path:"test" > file_type:FILE_TYPE_REGULAR url:"s3://tests3recursiveputs/test" recursive:true  529.741697ms <nil> test/29388.txt test/29388.txt}
```

I think the `protolion` package automatically does the marshaling but then I'd have to back the struct w a proto object. I almost [could've used their rpc error method](https://github.com/pachyderm/pachyderm/blob/master/src/server/vendor/go.pedge.io/proto/rpclog/protorpclog.go#L79) except for this new log type I needed a few new fields -- the `filePath` and objPath` info.

Comment (jdoliner) [inline src/server/pfs/server/api_server.go]: Actually you should just use the rpc logging method that's even better. `FilePath` and `ObjPath` fit nicely in that protobuf as the `File` field and the `URL` field. This way a recursive request will produce all the log messages that the individual requests would have produced.

Comment (sjezewski) [inline src/server/pfs/server/api_server.go]: Ah, cool. Yea I like that

**Comment by sjezewski** [comment]
Ok the new logs look like:

```
2017-02-27T21:30:57Z INFO  protorpclog.Call 
{
    "duration": "0.000s",
    "method": "putFileObj",
    "request": "file:<path:\"test/1876.txt\" > file_type:FILE_TYPE_REGULAR url:\"test/1876.txt\" recursive:true ",
    "service": "pfs.API"
}
```

And re-use the rpclog helpers.

**Comment by jdoliner** [comment]
Lovely

**Pull request merged**